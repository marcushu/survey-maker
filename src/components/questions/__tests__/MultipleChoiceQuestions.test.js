import React from 'react';
import {  render } from '@testing-library/react';
import MultipleChoiceQuestions from '../MultipleChoiceQuestion';

describe("Multiple Choice Questions", () => {
    const mockQuestion = {
        question: "mock question?",
        choices: ['choiceOne', 'choice_two', 'choice_Three']
    };

    const mockfunc = jest.fn();


    it("renders", () => {
        const { asFragment } = render(<MultipleChoiceQuestions
            question={mockQuestion}
            updateQuestion={mockfunc} />
            );

        expect(asFragment()).toMatchSnapshot();
    })

    it("correctly renders all of the labels", () => {
        const { getByText } = render(<MultipleChoiceQuestions
            question={mockQuestion}
            updateQuestion={mockfunc} />
            );

        expect(getByText("choiceOne")).toBeInTheDocument();
        expect(getByText("choice two")).toBeInTheDocument();
        expect(getByText("choice Three")).toBeInTheDocument();
    })

})