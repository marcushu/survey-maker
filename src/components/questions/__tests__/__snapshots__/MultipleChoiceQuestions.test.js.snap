// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Choice Questions renders 1`] = `
<DocumentFragment>
  <div
    class="listItem d-flex justify-content-between"
  >
    <div
      class="question"
    >
      mock question?
    </div>
    <div>
      <div
        class="form-check form-check-inline"
      >
        <input
          class="form-check-input"
          type="radio"
          value="choiceOne"
        />
        <label
          class="form-check-label"
          title=""
        >
          choiceOne
        </label>
      </div>
      <div
        class="form-check form-check-inline"
      >
        <input
          class="form-check-input"
          type="radio"
          value="choice_two"
        />
        <label
          class="form-check-label"
          title=""
        >
          choice two
        </label>
      </div>
      <div
        class="form-check form-check-inline"
      >
        <input
          class="form-check-input"
          type="radio"
          value="choice_Three"
        />
        <label
          class="form-check-label"
          title=""
        >
          choice Three
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
